internal/client/variables.go

```diff
@@ -0,0 +1,166 @@
+	_ = ctx
+	_ = filter
+
+	return nil, nil
+}
+
+// Get returns details for a variable by ID.
+func (c *VariablesClient) Get(ctx context.Context, variableID uuid.UUID) (*api.Variable, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, c.endpoint+"/variables/"+variableID.String(), http.NoBody)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+
+	resp, err := doRequest(c.hc, c.apiKey, req)
+	if err != nil {
+		return nil, fmt.Errorf("http error: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("status code %s", resp.Status)
+	}
+
+	var variable api.Variable
+	if err := json.NewDecoder(resp.Body).Decode(&variable); err != nil {
+		return nil, fmt.Errorf("failed to decode response: %w", err)
+	}
+
+	return &variable, nil
+}
+
+// GetByName returns details for a variable by name.
+func (c *VariablesClient) GetByName(ctx context.Context, name string) (*api.Variable, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, c.endpoint+"/variables/name/"+name, http.NoBody)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+
+	resp, err := doRequest(c.hc, c.apiKey, req)
+	if err != nil {
+		return nil, fmt.Errorf("http error: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("status code %s", resp.Status)
+	}
+
+	var variable api.Variable
+	if err := json.NewDecoder(resp.Body).Decode(&variable); err != nil {
+		return nil, fmt.Errorf("failed to decode response: %w", err)
+	}
+
+	return &variable, nil
+}
+
+// Update modifies an existing variable by ID.
+func (c *VariablesClient) Update(ctx context.Context, variableID uuid.UUID, data api.VariableUpdate) error {
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(&data); err != nil {
+		return fmt.Errorf("failed to encode data: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPatch, c.endpoint+"/variables/"+variableID.String(), &buf)
+	if err != nil {
+		return fmt.Errorf("error creating request: %w", err)
+	}
+
+	resp, err := doRequest(c.hc, c.apiKey, req)
+	if err != nil {
+		return fmt.Errorf("http error: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusNoContent {
+		return fmt.Errorf("status code %s", resp.Status)
+	}
+
+	return nil
+}
+
+// Delete removes a variable by ID.
+func (c *VariablesClient) Delete(ctx context.Context, variableID uuid.UUID) error {
+	req, err := http.NewRequestWithContext(ctx, http.MethodDelete, c.endpoint+"/variables/"+variableID.String(), http.NoBody)
+	if err != nil {
+		return fmt.Errorf("error creating request: %w", err)
+	}
+
+	resp, err := doRequest(c.hc, c.apiKey, req)
+	if err != nil {
+		return fmt.Errorf("http error: %w", err)
+	}
+	defer resp.Body.Close()
```

Comment:
we could probably wrap these in `doRequest()` and expose the method/path/params in the function sig.  not required though

Reply:
no, because defers run after the function returns, so that would close things early, e.g.

1. Call doRequest
2. defer resp.Body.Close()
3. Return from doRequest
4. Defer runs (body is closed)
5. We try to read from the body, but it's already closed